### PR TITLE
Add a Galaxy tool for TabPFN package (by Prof. Hutter's group)

### PR DESCRIPTION
**Pull request by anuprulez**
The PR adds a Galaxy tool that uses the TabPFN package (DOI: 10.48550/arXiv.2207.01848) to create a general-purpose classifier. 

- It takes input two files - train and test data along with their corresponding labels 
- TabPFNClassifier trains on the training data and predicts classes using test data
- Additionally, it outputs a precision-recall curve. 
- A conda package has been created using the TabPFN PyPi package (https://github.com/conda-forge/tabpfn-feedstock) because it is required by the Galaxy platform

Many more subsequent analyses can be added as required such as creating workflows with this tool. 

A few challenges:

- One challenge is the export this classifier to standard ONNX format. The standard export of Pytorch models to ONNX does not work with the trained TabPFN model. We may require it, otherwise, we may have to export the trained model as a zipped file.

- Additionally, to move the discussion around integrating the `tabpfn_client` package - it requires a separate login before use, so integrating it to Galaxy would be complicated I think unless there is a way to bypass the login process.

Thank you!

ping @bgruening 

tabpfn1

tabpfn2

tabpfn3

### CODE REVIEW

File: tools/tabpfn/main.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import time
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from sklearn.metrics import accuracy_score, average_precision_score, precision_recall_curve
+from tabpfn import TabPFNClassifier
+
+
+def separate_features_labels(data):
+    df = pd.read_csv(data, sep=",")
```
Comment (bgruening) [inline tools/tabpfn/main.py:14]: can we read tabular? or both ... but tabular would be nice.

File: tools/tabpfn/tabpfn.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            **Input files**
+            - Training data
+            - Test data
+
+            **Output files**
+            - Predicted data along with predicted labels
+        ]]>
+    </help>
+    <citations>
+        <citation type="doi">10.48550/arXiv.2207.01848</citation>
```
Comment (bgruening) [inline tools/tabpfn/tabpfn.xml:60]: we need to update this one

File: tools/tabpfn/tabpfn.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    <version_command>echo "@VERSION@"</version_command>
+    <command detect_errors="aggressive">
+    <![CDATA[
+        python '$__tool_directory__/main.py'
+            --train_data '$train_data'
+            --test_data '$test_data'
+    ]]>
+    </command>
+    <inputs>
+	<param name="train_data" type="data" format="csv" label="Train data" help="Please provide training data for training model."/>
```
Comment (bgruening) [inline tools/tabpfn/tabpfn.xml:26]: Some more hints how the data needs to be structured would be good.

**Comment by bgruening** [comment]
The results seem to be a bit random, so we need to work with asserts.

File: tools/tabpfn/tabpfn.xml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<tool id="tabpfn" name="Tabular data prediction using TabPFN" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
+    <description>with PyTorch</description>
+    <macros>
+        <token name="@TOOL_VERSION@">0.1</token>
```
Comment (bgruening) [inline tools/tabpfn/tabpfn.xml:4]:
```suggestion
        <token name="@TOOL_VERSION@">2.0.3</token>
```

**Review by bgruening — APPROVED** [review]
